xt/testapp/t/004_selenium.t

```diff
@@ -0,0 +1,312 @@
+
+=back
+
+=head2 Configuration
+
+Each IdP is set up in xt/testapp/IdPs/idp_name.  This script requires
+the existance of a configuration file named selenium.yml.
+
+The following fields are supported.  Not all IdPs need the full list
+of configuration fields.
+
+=over
+
+=item * L<username>
+
+The username to be entered into the IdP
+
+=item * L<username_fn>
+
+The identifier used to find the username field on the IdP login page
+
+=item * L<username_by>
+
+The method used to find the "username_fn" on the IdP login page. The supported
+methods are: class, class_name, css, id, link, link_text, partial_link_text,
+tag_name, name, xpath.
+
+=item * L<separate_passwd_page>
+
+Boolean that specifies whether the IdP has implemented the username and password
+as separate pages.
+
+=item * L<password>
+
+The password to be entered into the IdP
+
+=item * L<password_fn>
+
+The identifier used to find the password field on the IdP login page
+
+=item * L<password_by>
+
+The method used to find the "password_fn" on the IdP login page. The supported
+methods are: class, class_name, css, id, link, link_text, partial_link_text,
+tag_name, name, xpath.
+
+=item * L<login_btn>
+
+The identifier used to find the login button on th IdP's login page.  It is assumed
+that "clicking" the button will submit the username and/or password.
+
+Note that where there are separate username and password pages the same button
+identifier is assumed to be the same.
+
+=item * L<login_btn_by>
+
+The method used to find the "login_btn" on the IdP login page. The supported
+methods are: class, class_name, css, id, link, link_text, partial_link_text,
+tag_name, name, xpath.
+
+=item * L<post_login_page_title>
+
+Some IdPs provide a web page that is displayed after login and this allows the
+script to recoginize the page and move past it.
+
+=item * L<post_login_page_btn>
+
+The identifier used to find a button on the IdP's post login page.  It is assumed
+that "clicking" the button will move on to the NetSAML2 testapp's logged in page.
+
+=item * L<post_login_page_btn_by>
+
+The method used to find the "post_login_page_btn" on the IdP login page. The supported
+methods are: class, class_name, css, id, link, link_text, partial_link_text,
+tag_name, name, xpath.
+
+=item * L<post_logout_page_title>
+
+Some IdPs provide a web page that is displayed after logout and this allows the
+script to recoginize the page and move past it to the Net::SAML2 testapp login page.
+
+=item * L<login_bindings>
+
+Array of login bindings that the IdP supports (supported are: post, redirect).
+
+=item * L<logout_bindings>
+
+Array of logout bindings that the IdP supports (supported are: local, post, redirect).
+
+=item * L<issuer_value>
+
+Value of the issuer to find on the user logged in page for Net::SAML2 testapp to
+confirm that the login worked
+
+=back
+
+=cut
+if (! -d "$RealBin/selenium" ) {
+    warn "Created directory $RealBin/selenium\n" if (make_path("$RealBin/selenium"));
```

Comment:
Is this a warn or more some kind of informational message?

Reply:
informational - could the  print